config/env_spec_request.go

```diff
@@ -122,23 +122,28 @@ func (e *EnvironmentSpecRequest) GetParamValue(param APIOperationParameter) stri
 	var value string
 	switch m := param.Match.(type) {
 	case Header:
-		value = e.request.Attributes.Request.Http.Headers[strings.ToLower(string(m))]
+		key := strings.ToLower(string(m))
+		value = e.request.Attributes.Request.Http.Headers[key]
 		// Per Envoy: If multiple headers share the same key, they are merged per HTTP spec.
 		// So, we're just grabbing the first value (up to any comma).
 		if indx := strings.Index(value, ","); indx > 0 {
 			value = value[:indx]
 		}
+		log.Debugf("param from header %q: %q", key, value)
```

Comment:
Is it okay to print the key as is? Should we redact it?

Reply:
Yeah, I wondered about that myself. I sort of convinced myself that logging it only at debug level might be okay, but we really have no idea which parameters are sensitive. So while it will make debugging more difficult, perhaps we should not log any of these params - even at debug level.

Reply:
Otherwise, we could potentially severely truncate all the values to just a few characters.

Reply:
Also... this applies to header manipulation logging as well.

Reply:
Yeah truncating values sounds good.